java/src/main/java/io/cucumber/gherkin/GherkinTokenMatcher.java

```diff
@@ -180,26 +189,71 @@ private boolean match_DocStringSeparator(Token token, String separator, boolean
         return false;
     }
 
+    private void prepareStepLineLoopUnrolling() {
```

Comment:
The `currentDialect` is mutable. I don't see how this would work without some sort of cache invalidation.

Reply:
My mistake, it was done in another commit => I'll do a separate PR.